homeassistant/components/whirlpool/sensor.py

```diff
@@ -256,6 +256,8 @@ def __init__(
         )
         self._attr_has_entity_name = True
         self._attr_unique_id = f"{said}-{description.key}"
+        if self._attr_native_value is None:
+            self._attr_native_value = utcnow()
```

Comment:
The condition is not needed.
Also, please update the PR description to make it clear what problem this change solves.

Reply:
When the washer is running, the and the integration is initialized (without a stored value) this code fails to calculate:

https://github.com/home-assistant/core/blob/aec6af43046d35010efada702550d81a3d0bb3d4/homeassistant/components/whirlpool/sensor.py#L293-L303

and remains unknown until the washer/dryer completes a cycle.

Reviewing this a bit closer, I should put this test in https://github.com/home-assistant/core/blob/aec6af43046d35010efada702550d81a3d0bb3d4/homeassistant/components/whirlpool/sensor.py#L262 if the value is None, just to initialize it to "something" so that the calculation succeeds.

OR just put the test in the if running logic - so the first update from the API will put a time stamp in and it remains "unknown" until then.

Reply:
Went with the last "option"....

Reply:
Hoping to get this in 2023.4....